packages/std/src/math/isqrt.rs

```diff
@@ -12,7 +14,7 @@ where
     I: Unsigned
         + ops::Add<I, Output = I>
         + ops::Div<I, Output = I>
-        + ops::Shr<u8, Output = I>
+        + ops::Shr<u32, Output = I>
```

Comment:
Just out of curiosity, why this change? Is it because a `u8` wouldn't be able to specify a 256-bit right shift for a `Uint256`? Is that a valid right-shift in the first place? I'm asking because, when trying to right shift a `u32` by 32, I'm getting:
```rust
>> let n: u32 = 1;
>> n >> 32
attempt to shift right by `32_i32`, which would overflow
this arithmetic operation will overflow
>> 
```

(`checked_shr` works, though).

Reply:
Apparently, the choice of operand here is kind of arbitrary. I initially chose `u8` because I figured it's "big enough", but then discovered the std library tends to implement `u32` for these things - that's what I had to use for the[`u128::checked_shr`](https://doc.rust-lang.org/std/primitive.u128.html#method.checked_shr) method. Figured it's better to just stick to the convention than perform an extra conversion here.

So basically, it has nothing to do with whether `256` would fit or not.

Reply:
I guess  `u32` just looks more like a "numerical type" than `u8`; or even, `u16`, that for me at least is still somewhat related to bytes.